### PR TITLE
Pass children prop to MenuItem and textComponent to MenuCollapsibleItem

### PR DESCRIPTION
**Pull request by MagdaxMai**
This adds a children property to MenuItem and a textComponent property to MenuCollapsibleItem to allow rendering components instead of strings for text. This is needed to render FormattedMessage components for instance.

### CODE REVIEW

**Comment by dkarnutsch** [comment]
We keep text to not introduce a breaking change?

With both text and children being optional you might not have one of them...

I would suggest removing text completely, making it a breaking change and changing its target to next.

File: packages/admin/src/mui/menu/CollapsibleItem.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ export interface ICollapsibleItemProps extends IMenuItemProps {
     collapsible: boolean;
     secondaryAction?: React.ComponentType<ICollapsibleItemSecondaryActionProps>;
     isOpen?: boolean;
+    textComponent?: React.ReactNode;
```
Comment (nsams) [inline packages/admin/src/mui/menu/CollapsibleItem.tsx:20]: text, not textComponent

File: packages/admin/src/mui/menu/Item.tsx [hunk 2]
```diff
@@ -5,26 +5,27 @@ import { IMenuLevel, MenuContext } from "./index";
 import * as sc from "./Item.sc";
 
 export interface IMenuItemProps extends IMenuLevel {
-    text: string;
+    text?: string;
```
Comment (nsams) [inline packages/admin/src/mui/menu/Item.tsx:8]: could also be React.ReactNode

File: packages/admin/src/mui/menu/Item.tsx [hunk 3]
```diff
@@ -5,26 +5,27 @@ import { IMenuLevel, MenuContext } from "./index";
...
     icon?: React.ReactElement;
     secondaryAction?: React.ReactNode;
     enableMargin?: boolean;
+    children?: React.ReactNode;
```
Comment (nsams) [inline packages/admin/src/mui/menu/Item.tsx:12]: I'm unsure if having text PLUS children for the same thing is a good idea (if we keep text just for compatibility, it should be deprecated)

Comment (MagdaxMai) [reply packages/admin/src/mui/menu/Item.tsx:11]: the text property has now been completely removed

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,13 @@
 
 All notable changes to this project will be documented in this file. This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Next
+
+### Incompatible Changes
+
+-   Removed `text` property from `MenuItem`
```
Comment (dkarnutsch) [inline CHANGELOG.md:9]:
```suggestion
-   Removed `text` property from `MenuItem` in favor of more flexibel `children`. 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,13 @@
...
+-   `children` can now be passed to `MenuItem` to render text
```
Comment (dkarnutsch) [inline CHANGELOG.md:10]:
```suggestion
```

**Review by dkarnutsch — CHANGES_REQUESTED** [review]

File: packages/admin-stories/src/admin/mui/Menu.tsx [hunk 6]
```diff
@@ -3,16 +3,25 @@ import { Typography } from "@material-ui/core";
...
-            <MenuItemRouterLink text="Foo5" icon={<Home />} to="/foo5" />
+        <MenuItemRouterLink  icon={<CalendarToday />} to="/foo1" >
+            <FormattedMessage id="comet.stories.foo1" defaultMessage="Foo1"/>
+        </MenuItemRouterLink>
+        <MenuItemRouterLink icon={<School />} to="/foo2" >
+            <FormattedMessage id="comet.stories.foo2" defaultMessage="Foo2"/>
+        </MenuItemRouterLink>
+        <MenuCollapsibleItem text={<FormattedMessage id="comet.stories.foo3" defaultMessage="Foo3"/>} icon={<Settings />} collapsible={true} isOpen={false}>
+            <MenuItemRouterLink icon={<Home />} to="/foo4" >
+                <FormattedMessage id="comet.stories.foo4" defaultMessage="Foo4"/>
```
Comment (nsams) [inline packages/admin-stories/src/admin/mui/Menu.tsx:20]: When looking at this it feels not right: MenuCollapsibleItem has a text prop, MenuItemRouterLink children. Using text for both would be more consistent - what do you think?

**Comment by nsams** [comment]
@MagdaxMai can you finish this?

File: packages/admin-stories/src/admin/mui/Menu.tsx (hunk 6 above)
Comment (MagdaxMai) [reply packages/admin-stories/src/admin/mui/Menu.tsx:20]: yes i agree with you, i will adjust it

**Review by dkarnutsch — APPROVED** [review]

**Comment by nsams** [comment]
please merge/rebase to current next

and your change isn't incompatible anymore

**Comment by jamesricky** [comment]
@MagdaxMai @nsams I think this is no longer necessary, the `text: string;` prop was replaced with `primary: React.ReactNode;` and `secondary?: React.ReactNode;` [here](https://github.com/vivid-planet/comet-admin/commit/43b57fb3cd99bb82aae6307fb65d3aa111e85289#diff-6249f1163fc456b9c49e2ef50a742506ad3ba9899455cfb85574540b6f69167cR21-R22).

**Comment by nsams** [comment]
> @MagdaxMai @nsams I think this is no longer necessary

ooops :) thanks for the hint!

**Pull request closed without merging**